oap-server/server-receiver-plugin/skywalking-meter-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/meter/provider/process/EvalSingleData.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.receiver.meter.provider.process;
+
+import lombok.Data;
+import org.apache.skywalking.apm.network.language.agent.v3.Label;
+import org.apache.skywalking.apm.network.language.agent.v3.MeterSingleValue;
+
+import java.math.BigDecimal;
+import java.util.stream.Collectors;
+
+/**
+ * Support counter, gauge
+ */
+@Data
+public class EvalSingleData extends EvalData<EvalData> {
+
+    private double value;
+
+    public static EvalSingleData build(MeterSingleValue value, MeterProcessor processor) {
+        final EvalSingleData singleEvalData = new EvalSingleData();
+        singleEvalData.name = value.getName();
+        singleEvalData.labels = value.getLabelsList().stream()
+            .collect(Collectors.toMap(Label::getName, Label::getValue));
+        singleEvalData.processor = processor;
+        singleEvalData.value = value.getValue();
+        return singleEvalData;
+    }
+
+    @Override
+    public EvalData multiply(double value) {
+        return copyTo(EvalSingleData.class, instance -> instance.value = this.value * value);
+    }
+
+    @Override
+    public EvalData multiply(EvalData data) {
+        if (data instanceof EvalHistogramData) {
+            throw new IllegalArgumentException("Only support multiply from single value");
+        }
```

Comment:
From the exception message `Only support multiply from single value`, the condition should be `if (!(data instanceof EvalSingleData))` so that it applies when `data instanceof EvalMultipleData` or `data instanceof` other future subclass of `EvalData`

Reply:
Thanks, that's a good suggest.